plutus-ir/test/datatypes/listMatch.plc

```diff
@@ -0,0 +1,22 @@
+(let (rec)
+  (datatypebind
+    (datatype
+      (tyvardecl List (fun (type) (type)))
+      (tyvardecl a (type))
+      match_List
+      (vardecl Nil [List a]) (vardecl Cons (fun a (fun [List a] [List a])))
+    )
+  )
+  [
+    [
+      {
+        [{match_List (all a (type) (fun a a))} {Nil (all a (type) (fun a a))}]
```

Comment:
Since we're writing these out by hand now, I think it's probably worth declaring unit in many of them to avoid the proliferation of `forall a . a -> a`